mlx/backend/common/qrf.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright © 2023 Apple Inc.
+
+#include <cassert>
+
+#include <iostream>
+#include "mlx/backend/common/copy.h"
+#include "mlx/device.h"
+#include "mlx/dtype.h"
+#include "mlx/ops.h"
+#include "mlx/primitives.h"
+#include "mlx/utils.h"
+
+#ifdef ACCELERATE_NEW_LAPACK
+#include <vecLib/lapack.h>
+#include <vecLib/lapack_types.h>
+#endif
+
+namespace mlx::core {
+
+template <typename T>
+struct lpack;
+
+template <>
+struct lpack<float> {
```

Comment:
not sure if we need an implementation for fp64 here because mlx doesn't support it?
for dtype complex64 we could use CGEQRF and CORMQR.

cc @awni

Reply:
We don't need to support fp64 yet. If there is complex64 that would be nice to add. For other floating point types (fp16 and bf16), I would just through for now.

Reply:
makes sense